### ISSUE TITLE
[PDKI-1144] VHV Traffix light system POC - llm-as-judge

### ISSUE THREAD

**Issue by jinijohn26** [5Plus-Test/jiramigrationtest#63]
> Migrated from Jira: [PDKI-1144](https://pentadoc.atlassian.net/browse/PDKI-1144)

---
### Checklist

- [x] brainstorm a more precise llm-as-judge prompts and approach
- [x] evaluate the judge on our GT
- [x] evaluate the judge on VHV GT
- [x] share results with vhv

---
### Comments

**Boris Loktev** (30/Dec/25 2:05 PM):
> h2. LLM-as-Judge Implementation Update

I've completed the initial implementation of the LLM-as-Judge system in the {{bedrock_judge/}} module. Here's an overview:

h3. Architecture

*Core Service*: {{BedrockJudgeService}} uses Claude Haiku 4.5 via AWS Bedrock with extended thinking mode to evaluate extraction quality.

*Two Evaluation Judges*:

# *General Info Judge* - Evaluates personal info, weight, occupation, hobbies extraction
# *Medical Conditions Judge* - Evaluates medical condition extraction completeness and accuracy

Both judges run in parallel via {{asyncio.gather()}} for performance.

h3. Quality Metrics (6 signals per extraction type)

||Metric||Range||Purpose||
|{{completeness_confidence}}|0 or 1|Binary: did we capture everything?|
|{{consistency_score}}|0.0-1.0|Internal logical coherence (BMI math, timeline logic)|
|{{source_attribution_quality}}|0.0-1.0|Quality of document citations|
|{{uncertainty_flags}}|array|12 specific flags (e.g., {{symptom_details_incomplete}})|
|{{reasoning}}|string|Brief explanation of issues found|
|{{suggested_missing_items}}|array|Actionable feedback with doc page references|

h3. Training Pipeline (for calibrator model)

Complete ML pipeline to predict extraction correctness:

# *Synthetic Data Generation* ({{generate_synthetic_data.py}}) - Creates corrupted versions of ground truth with severity levels (clean/minor/moderate/major)
# *Judge Signal Collection* ({{run_judge_on_synthetic.py}}) - Runs judge on synthetic data
# *Feature Extraction* ({{create_training_dataset.py}}) - Extracts 17 features (2 binary completeness + 3 continuous scores + 12 uncertainty flags)
# *Calibrator Training* ({{train_calibrator.py}}) - Supports XGBoost, Random Forest, Logistic Regression with cross-validation

h3. Initial Results (VHV Random Forest Model)

* *Test Set AUC-ROC*: 0.923
* *Test Set PR-AUC*: 0.875
* *Top predictive features*: {{medical_conditions_consistency_score}} and {{general_info_consistency_score}}

h3. Multi-Format Support

The judge supports both:

* *Bedrock format* (simplified structure with boolean KO, array measurements)
* *Claude format* (detailed structure with full KO objects, source arrays)

h3. Next Steps

* [ ] Fine-tune prompt for better consistency scoring calibration
* [ ] Increase training dataset size for calibrator
* [ ] Integrate calibrator into extraction pipeline for traffic light system

h3. Usage

{code:bash}# Run judge evaluation
uv run bedrock_judge/run_judge_evaluation.py \
    --test-cases evaluation/test_cases \
    --output bedrock_judge/results \
    --service bedrock{code}

Code is in branch {{feat/add_aws_bedrock_inference}}.

**Boris Loktev** (13/Jan/26 8:18 PM):
> all the results will be presented tomorrow in a meeting
